lib/data/demos.dart

```diff
@@ -49,22 +50,56 @@ import 'package:gallery/themes/material_demo_theme_data.dart';
 
 const _docsBaseUrl = 'https://api.flutter.dev/flutter';
 
+enum GalleryDemoCategory {
+  study,
+  material,
+  cupertino,
+  other,
+}
+
+extension GalleryDemoExtension on GalleryDemoCategory {
+  String get name => describeEnum(this);
+
+  String displayTitle(GalleryLocalizations localizations) {
+    switch (this) {
+      case GalleryDemoCategory.material:
+        return 'MATERIAL';
+      case GalleryDemoCategory.cupertino:
+        return 'CUPERTINO';
+      case GalleryDemoCategory.other:
+        return localizations.homeCategoryReference;
+      case GalleryDemoCategory.study:
+    }
+    return null;
+  }
+}
+
 class GalleryDemo {
   GalleryDemo({
     @required this.title,
-    @required this.slug,
-    @required this.icon,
+    @required this.category,
     @required this.subtitle,
-    @required this.configurations,
+    // Parameters below are required for non-study demos.
+    this.slug,
+    this.icon,
+    this.configurations,
   })  : assert(title != null),
-        assert(icon != null),
-        assert(configurations != null && configurations.isNotEmpty);
+        assert(category != null),
+        assert(subtitle != null),
+        assert(category == GalleryDemoCategory.study ||
+            (slug != null &&
+                icon != null &&
+                configurations != null &&
+                configurations.isNotEmpty));
```

Comment:
Formatting looks strange here but I guess we can't do much about this

Reply:
Yeah, I tried to get it to look better but didn't find something

Reply:
Yeah, I tried to get it to look better but didn't find something